2.14/linux/argbash-templates/set-guest-attributes.m4

```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
+
+_default_config_dir="${APP_HOME}/etc"
+_default_profiles_json="${APP_HOME}/etc/ws-security/profiles.json"
+_default_in_place_editing=false
+_default_hostname=$HOSTNAME
+
+# m4_ignore(
+echo "This is just a script template, not the script (yet) - pass it to 'argbash' to fix this." >&2
+exit 11  #)Created by argbash-init v2.7.1
+# ARG_OPTIONAL_SINGLE([config-directory],[c],[location where the config files are],[${_default_config_dir}])
+# ARG_OPTIONAL_SINGLE([profiles-json],[j],[JSON file with profile attributes],[${_default_profiles_json}])
+# ARG_OPTIONAL_SINGLE([hostname],[h],[hostname],[${_default_hostname}])
+# ARG_OPTIONAL_BOOLEAN([in-place],[i],[replace original config files with edits],[off])
+# ARG_POSITIONAL_SINGLE([profile],[p],[security profile to use])
+# ARG_DEFAULTS_POS
+# ARG_HELP([<The general help message of my script>])
+# ARGBASH_GO
+
+# [ <-- needed because of Argbash
+
+trap "exit 1" TERM
+export TOP_PID=$$
+
+HOSTNAME=""
+PROFILE=""
+PROFILE_FILE=""
+USER_ATTRIBUTES_FILE=""
+ATTRIBUTES_WORKING_FILE=""
+CONFIG_WORKING_FILE=""
+CONFIG_DIR=""
+IN_PLACE_EDITING=""
+
+# Handles printing error messages to the terminal and exiting the program.
+# ${1} - String of the error message to print
+function error() {
+    if [[ -f $ATTRIBUTES_WORKING_FILE ]]; then
```

Comment:
Looks like these first few lines could be replaced with a call to the cleanup function

Reply:
I made a change to `clean_up()` to have a parameter for handling calls from the error function, which removes the repetitive code.

Reply:
👍